sdk/python/kfp/compiler/compiler.py

```diff
@@ -189,31 +190,33 @@ def _create_pipeline(
 
         return pipeline_spec
 
-    def _write_pipeline_spec_json(
+    def _write_pipeline_spec_yaml(
         self,
         pipeline_spec: pipeline_spec_pb2.PipelineSpec,
         output_path: str,
     ) -> None:
-        """Writes pipeline spec into a json file.
+        """Writes pipeline spec into a YAML file.
 
         Args:
             pipeline_spec: IR pipeline spec.
             ouput_path: The file path to be written.
 
         Raises:
-            ValueError: if the specified output path doesn't end with the
-                acceptable extention.
+            ValueError: if the specified output path doesn't end with a
+                .yaml extention.
         """
-        json_text = json_format.MessageToJson(pipeline_spec, sort_keys=True)
 
-        if output_path.endswith('.json'):
-            with open(output_path, 'w') as json_file:
-                json_file.write(json_text)
-        else:
+        if not output_path.endswith(('.yaml', ".yml")):
```

Comment:
Maybe check for .json extension here as well and produce JSON file if it's requested to ensure backward compat?

Reply:
Done!